networkx/convert_matrix.py

```diff
@@ -1360,3 +1364,174 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
     return from_numpy_matrix(
         A, parallel_edges=parallel_edges, create_using=create_using
     )
+
+
+def generate_node_dataframe(
+    G: nx.Graph,
+    funcs: List[Callable],
+):
+    """
+    Return a pandas DataFrame representation of nodes and their metadata.
+
+    This function accepts a graph and a collection of user-defined functions
+    and returns a pandas DataFrame in which the nodes' are the DataFrame index,
+    with column names and DataFrame values being arbitrarily defined
+    based off user functions.
+
+    ``funcs`` is an iterable of callables, each of which has the signature
+
+        f(n, d) -> pd.Series
+
+    In other words, it accepts a (node, node attribute dict) pair,
+    such that `n` is the node,
+    and `d` is the node attribute dictionary.
+    Each function ``f`` must return a pandas Series whose name is the node.
+
+    As an example:
+
+    .. code-block:: python
+
+        def x_vec(n: Hashable, d: Dict[Hashable, Any]) -> pd.Series:
+            return pd.Series({"x_coord": d["x_coord"]}, name=n)
+
+    One fairly strong assumption is that each func
+    has all the information it needs to act
+    stored on the attribute dictionary,
+    and does not take in any other keyword arguments.
+
+    If you need to reference an external piece of information,
+    such as a dictionary to look up values,
+    set up the function to accept the dictionary,
+    and use ``functools.partial``
+    to "reduce" the function signature to just ``(n, d)``.
+    An example below:
+
+    .. code-block:: python
+
+        from functools import partial
+        def get_molweight(n, d, mw_dict):
+            return pd.Series({"mw": mw_dict[d["amino_acid"]]}, name=n)
+
+        mw_dict = {"PHE": 165, "GLY": 75, ...}
+        get_molweight_func = partial(get_molweight, mw_dict=mw_dict)
+
+        generate_feature_dataframe(G, [get_molweight_func])
```

Comment:
Can ```lambda``` functions work too?  
Like  ```generate_feature_dataframe(G, [lambda n, d:get_molweight(n, d, mw_dict)])```

I assume so... is it worth saying?

Reply:
Yes! I've used the same pattern too 😄.